src/ng/animate.js

```diff
@@ -416,8 +416,8 @@ var $AnimateProvider = ['$provide', /** @this */ function($provide) {
        * @param {Promise} animationPromise The animation promise that is returned when an animation is started.
        */
       cancel: function(runner) {
-        if (runner.end) {
-          runner.end();
+        if (runner.cancel) {
+          runner.cancel();
```

Comment:
I still don't understand why it worked before but not now. Previously, `.then(cb)` was used. This is implemented in `$animate` like this (code changed for brevity):

```js
then: resolveHandler => {
  return $q((resolve, reject) => this.done(status => status === false ? reject() : resolve())).
            then(resolveHandler)
}
```

So, afaict we should be able to safely replace `.then(cb)` with `.done(status => status !== false && cb())` keeping the exact same behavior minus the extra digest (and asynchronicity) of `$q`.

Is the fact that the callback is now synchronously called causing the problem?